### PR TITLE
Add support for the HTTP PATCH method.

### LINKED ISSUE #20: Support PATCH HTTP Method
Used by backbone.js for partial object updates.

### PR DESCRIPTION
**Pull request by nesv**
@cespare I messed something up big-time, and it's been a long day, so I just split the changes off into their own branch. They have been squashed, like you asked.

### CODE REVIEW

**Comment by nesv** [comment]
Replaces #26

File: mux_test.go [hunk 1]
```diff
@@ -207,3 +207,36 @@ func newRequest(method, urlStr string, body io.Reader) *http.Request {
 	}
 	return req
 }
+
+func TestMethodMatch(t *testing.T) {
```
Comment (cespare) [inline mux_test.go]: Should this be `TestMethodPatch`?

Comment (cespare) [inline mux_test.go]: Can you move this test up above `func newRequest`?

File: mux_test.go [hunk 2]
```diff
@@ -207,3 +207,36 @@ func newRequest(method, urlStr string, body io.Reader) *http.Request {
...
+	p := New()
+	p.Patch("/foo/bar", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
+
+	// Test to see if we get a 405 Method Not Allowed errors from trying to
+	// issue a GET request to a handler that only supports the PATCH method.
+	req, err := http.NewRequest("GET", "/foo/bar", nil)
```
Comment (cespare) [inline mux_test.go]: Since there's a new helper, here and below you can just write `req := newRequest("GET", "/foo/bar", nil)`

**Comment by cespare** [comment]
@nesv No worries! And thanks for updating the test.

Looking over this again, I noticed a couple of other small things.

**Comment by cespare** [comment]
Actually, I just merged it and made the fixes myself. Thanks again!

**Pull request merged**